src/main/kotlin/biokotlin/seqIO/FastqIO.kt

```diff
@@ -132,5 +135,29 @@ $filename  Error: ${e.message}""")
         }
 
     }
+}
+
+fun writeFastq(input: Collection<SeqRecord>, filename: String) {
+    val extension = File(filename).extension
+    val newFilename = if (extension.equals("fq", false) || extension.equals("fastq", false)) {
+        filename
+    } else {
+        "$filename.fq"
+    }
+
+    println("FastqIO: writeFastq: writing file $newFilename")
 
+    bufferedWriter(newFilename).use { outputFile ->
+        for (record in input) {
+            val seq = when (record) {
+                is NucSeqRecord -> record.sequence.toString()
+                else -> throw IllegalStateException("writeFastq trying to output something other than Nuc Seq.")
+            }
+            outputFile.write("${record.id}\n")
+            outputFile.write("${seq}\n")
+            outputFile.write("+\n")
+            outputFile.write(record.letterAnnotations["quality"]?.joinToString("")?:"")
```

Comment:
This looks like the code writes an empty string if "letterAnnotations" is empty, which would be illegal for a fastQ file format.  In reality, the SeqRecord class has a "require" that verifies the annotation size matches the sequence size.

Consider added a 1 line command that indicates SeqRecord object ensures quality annotation size is the same as sequence size.

Reply:
Done. Thank you!